### PR TITLE
Feat/Improve progression screen with achievements linked to social interactions

### PR DESCRIPTION
**Pull request by alvaro1080**
## Overview
The objective of this pull request is to improve the progression screen with new achievements that are linked to social interactions between the users. As this is the main objective of our app, we wanted to reward the users who adds new friends with the bluetooth feature with dedicated achievements.

## Changes
- A new function `checkFriends` in the viewModel, in order to reward the user who added friends
- Achievements based on the number of friends
- Removed parts of progression screen (for instance the parks visited metric) that are not yet implemented, in order to aim for completeness in the screen

## How to Test

1. Add a friend
2. Go to the achievements part of the progression screen

You should have a new achievements for adding your first friend!

## Screenshot

image

### CODE REVIEW

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 1]
```diff
@@ -192,6 +171,40 @@ fun ProgressScreen(
   }
 }
 
+@Composable
+fun AchievementColumn(achievements: List<String>) {
+
+  if (achievements.isEmpty()) {
+
+    Text(
+        text = "You don't have any achievements yet!",
```
Comment (misterM125) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:180]: Suggestion: As per the coach feedback, all hardcoded strings will have to be moved to the `strings.xml`, might be nice to move it now to make refactoring easier later

File: app/src/androidTest/java/com/android/streetworkapp/ui/Progression/ProgressionTest.kt [hunk 2]
```diff
@@ -84,7 +85,7 @@ class ProgressionTest {
     composeTestRule.onNodeWithTag("percentageInsideCircularProgressBar").assertIsDisplayed()
     composeTestRule.onNodeWithTag("scoreTextUnderCircularProgressBar").assertIsDisplayed()
     composeTestRule.onNodeWithTag("metricCardScoreValue").assertIsDisplayed()
-    composeTestRule.onNodeWithTag("metricCardParksVisitedValue").assertIsDisplayed()
+    // composeTestRule.onNodeWithTag("metricCardParksVisitedValue").assertIsDisplayed()
```
Comment (misterM125) [inline app/src/androidTest/java/com/android/streetworkapp/ui/Progression/ProgressionTest.kt:88]: comment: if this assert will not be used anymore, might as well delete it

**Review by misterM125 — APPROVED** [review]
Great PR! The addition of achievements for the number of friends a user has is a great incentive to socialize and a nice addition to the gamification of the app.
There are just a couple minor corrections that i can suggest but i will approve the PR since those are very small

File: app/src/androidTest/java/com/android/streetworkapp/ui/Progression/ProgressionTest.kt (hunk 2 above)
Comment (alvaro1080) [reply app/src/androidTest/java/com/android/streetworkapp/ui/Progression/ProgressionTest.kt:88]: Done, thank you!

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 1 above)
Comment (alvaro1080) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt]: You are right, it is done thank you!

**Pull request merged**